File: C/8_myAtoi.c
int myAtoi(char* str) {
    int val = 0;
    int sign = 1;
    int old_val = 0;
    
    while(isspace(*str))
        *str++;
    
    if (*str == '-' || *str == '+') {
        if (*str == '-') sign = -1;
        *str++;
    }
    
    while(isdigit(*str)) {
        old_val = val;
        val = val * 10 + *str - '0';
        if(val / 10 != old_val) {
            return sign == 1 ? INT_MAX : INT_MIN;
        }
        
        *str++;
    }
    
    return sign * val;
}

File: python/8_myAtoi.py
class Solution(object):
    def myAtoi(self, str):
        """
        :type str: str
        :rtype: int
        """
        s = str.strip()
        if len(s) == 0:
            return 0

        MAX_INT = 2**31 - 1
        MIN_INT = -2**31
        
        sign = 1
        i = 0
        if s[0] in "-+":
            if (s[0] == '-'):
                sign = -1;
            i = 1
        
        val = 0
        for i in xrange(i, len(s)):
            if s[i].isdigit():
                val = val * 10 + ord(s[i]) - ord('0')
            else:
                break

        val = sign * val
        
        if val > MAX_INT:
            return MAX_INT
        elif val < MIN_INT:
            return MIN_INT
        else:
            return val
